app/client/components/user/createVendor.jsx

```diff
@@ -0,0 +1,357 @@
+import React  from 'react';
+import { connect } from 'react-redux';
+import styled from "styled-components";
+
+import AsyncButton from '../AsyncButton.jsx'
+import QrReadingModal from "../qrReadingModal.jsx"
+import { createUser, RESET_CREATE_USER } from '../../reducers/userReducer'
+import { StyledButton, ErrorMessage, ModuleHeader } from '../styledElements'
+
+class CreateVendorForm extends React.Component {
```

Comment:
left the way create vendor works alone for comparison, but did fix the logic around one_time_code and accidental globals here also